press-sync.php

```diff
@@ -702,6 +709,97 @@ public function approve_localhost_urls( $approve, $host, $url ) {
+	 * @param array   $settings     The Press Sync settings.
+	 * @param integer $next_page    Pagination counter for bath queries.
+	 */
+	public function sync_batch( $content_type = 'post', $settings = array(), $next_page = 1 ) {
+
+		$remote_domain     = isset( $settings['remote_domain'] ) ? $settings['remote_domain'] : '';
+		$press_sync_key    = isset( $settings['remote_press_sync_key'] ) ? $settings['remote_press_sync_key'] : '';
+		$sync_method       = isset( $settings['sync_method'] ) ? $settings['sync_method'] : cmb2_get_option( 'press-sync-options', 'sync_method' );
+		$objects_to_sync   = $content_type ? $content_type : cmb2_get_option( 'press-sync-options', 'objects_to_sync' );
+		$duplicate_action  = isset( $settings['duplicate_action'] ) ? $settings['duplicate_action'] : cmb2_get_option( 'press-sync-options', 'duplicate_action' );
+		$force_update      = isset( $settings['force_update'] ) ? $settings['force_update'] : cmb2_get_option( 'press-sync-options', 'force_update' );
+
+		// Initialize the connection credentials.
+		$this->init_connection( $remote_domain );
+
+		$prepare_object    = ! in_array( $objects_to_sync, array( 'attachment', 'comment', 'user' ), true ) ? 'post' : $objects_to_sync;
+		$wp_object         = in_array( $objects_to_sync, array( 'attachment', 'comment', 'user' ), true ) ? ucwords( $objects_to_sync ) . 's' : get_post_type_object( $objects_to_sync );
+		$wp_object         = isset( $wp_object->labels->name ) ? $wp_object->labels->name : $wp_object;
+
+		// Build out the url.
+		$url               = cmb2_get_option( 'press-sync-options', 'connected_server' );
+		$press_sync_key    = cmb2_get_option( 'press-sync-options', 'remote_press_sync_key' );
+		$url               = untrailingslashit( $url ) . '/wp-json/press-sync/v1/sync?press_sync_key=' . $press_sync_key;
+
+		// Prepare the remote request args.
+		$sync_args['duplicate_action']  = $duplicate_action;
+		$sync_args['force_update']      = $force_update;
+		$sync_args['objects_to_sync']   = $prepare_object;
+
+		// Prepare the correct sync method.
+		$sync_class        = 'prepare_' . $prepare_object . '_args_to_sync';
+		$total_objects     = $this->count_objects_to_sync( $objects_to_sync );
+		$taxonomies        = get_object_taxonomies( $objects_to_sync );
+		$next_page         = isset( $_POST['paged'] ) ? (int) esc_attr( $_POST['paged'] ) : $next_page;
+		$objects           = $this->get_objects_to_sync( $objects_to_sync, $next_page, $taxonomies );
+		$logs              = array();
+
+		// Prepare each object to be sent to the remote server.
+		foreach ( $objects as $key => $object ) {
+			$sync_args['objects'][ $key ] = $this->$sync_class( $object );
+		}
+
+		$logs = $this->send_data_to_remote_server( $url, $sync_args );
+
+		return array(
+			'objects_to_sync'         => $wp_object,
+			'total_objects'           => $total_objects,
+			'total_objects_processed' => ( $next_page * 5 ) - ( 5 - count( $objects ) ),
```

Comment:
I'm a little hazy on this math - why are you subtracting the count of objects from 5? If the number of objects is `> 5` then won't this end up adding `($next_page * 5)` to the negative number result from the right-hand of the equation?

Reply:
The number of objects will never be greater than 5 because the batch is hard coded to only pull in a max of 5 records.

Reply:
Gotcha, works for me!